src/modules/connection.js

```diff
@@ -1,99 +1,126 @@
-    return Promise.resolve(connection.chann);
+    return connection.conn;
   }
 
-  connection.chann = connection.conn.createChannel()
-    .then((channel) => {
-      channel.prefetch(prefetch);
+  /**
+   * Create the channel on the broker, once connection is successfuly opened.
+   * Since RabbitMQ advise to open one channel by process and node is mono-core, we keep only 1 channel for the whole connection.
+   * @return {Promise} A promise that resolve with an amqp.node channel object
+  */
+  getChannel() {
+    const url = this._config.host;
+    const prefetch = this._config.prefetch;
+    const connection = this.connections[url];
+
+    // cache handling, if channel already opened, return it
+    if (connection && connection.chann) {
+      return Promise.resolve(connection.chann);
+    }
+
+    connection.chann = connection.conn.createChannel()
+      .then((channel) => {
+        channel.prefetch(prefetch);
+
+        // on error we remove the channel so the next call will recreate it (auto-reconnect are handled by connection users)
+        channel.on('close', () => { delete connection.chann; });
+        channel.on('error', this._config.transport.error);
+
+        connection.chann = channel;
+        return channel;
+      });
+    return connection.chann;
+  }
 
-      //on error we remove the channel so the next call will recreate it (auto-reconnect are handled by connection users)
-      channel.on('close', () => { delete connection.chann; });
-      channel.on('error', this.config.transport.error);
+  /**
+   * Connect to AMQP and create channel
+   * @return {Promise} A promise that resolve with an amqp.node channel object
+   */
+   /* eslint arrow-body-style: "off" */
+  get() {
+    return this.getConnection().then(() => { return this.getChannel(); });
+  }
 
-      connection.chann = channel;
-      return channel;
+  /**
+   * Register an event on the amqp.node channel
+   * @param {string} on   the channel event name to be binded with
+   * @param {function} func the callback function to execute when the event is called
+   */
+  addListener(on, func) {
+    this.get().then((channel) => {
+      channel.on(on, func);
     });
+  }
 
-  return connection.chann;
-}
+  get config() {
+    return this._config;
+  }
 
-/**
- * Connect to AMQP and create channel
- * @return {Promise} A promise that resolve with an amqp.node channel object
- */
-function get() {
-  return getConnection.call(this).then(() => getChannel.call(this));
+  set config(value) {
+    this._config = value;
+  }
 }
 
-/**
- * Register an event on the amqp.node channel
- * @param {string} on   the channel event name to be binded with
- * @param {function} func the callback function to execute when the event is called
- */
-function addListener(on, func) {
-  this.get().then((channel) => {
-    channel.on(on, func);
-  });
-}
+
+let instance;
 
 module.exports = (config) => {
-  assert(config.hostname);
+  if (!instance && !config) {
+    return undefined;
```

Comment:
I afraid about this, maybe use `assert` to get a proper error message. so we can understand what happen.